src/AspNet.Security.OAuth.Yahoo/YahooAuthenticationHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Framework.Internal;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Yahoo {
+    public class YahooAuthenticationHandler : OAuthAuthenticationHandler<YahooAuthenticationOptions> {
+        public YahooAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {}
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            // Extract the profile object from the result payload
+            var profile = payload.Value<JObject>("profile");
+
+            // Extract all the claims from the profile
+            if (profile != null) {
```

Comment:
Have you tried moving this null check directly to the `YahooAuthenticationHelper` class?
Something like:

```csharp
public static string GetFamilyName([NotNull] JObject user) => user.Value<string>("familyName"); 
```

->


```csharp
public static string GetFamilyName([NotNull] JObject user) => user.Value<JObject>("profile")
                                                                 ?.Value<string>("familyName"); 
```

This way, one can use the helper without having to manually extract the profile node first :+1: